package de.dhbw.exercise;

import de.dhbw.exercise.visibility.person;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static de.dhbw.commons.Assertions.assertEquals;

public class VisibilityExerciseTest {

    @Test
    @DisplayName("""
        Übung 1:
        a) Implementiere eine Klasse 'Person' mit den Attributen
         - Name
         - Alter
        b) Entscheide selbst über die Datentypen der Attribute, aber mache sie "nicht öffentlich".
        c) Erstelle zudem Methoden für das Auslesen und Setzen der Attribute (get-/setName, get-/setAge)
        d) Schreibe hier einen einfachen Test: Setze konkrete Werte für Name und Alter (set), überprüfe dann,
           ob die Werte korrekt gesetzt wurden (get).
        
        Achte bewusst auf die Sichtbarkeiten der Attribute und Methoden und erkläre diese!
        """)
    public void exercise1() {
        // given
        person testPerson = new person();
        String geplanterName = "Lukas";
        int geplantesAlter = 19;

        // when
        testPerson.setName(geplanterName);
        testPerson.setAge(geplantesAlter);
        // then
        assertEquals(geplanterName, testPerson.getName());
        assertEquals(geplantesAlter, testPerson.getAge());

    }

}